sound/soc/sof/intel/hda-dai.c

```diff
@@ -466,18 +468,28 @@ static int ipc4_hda_dai_trigger(struct snd_pcm_substream *substream,
 
 	w = snd_soc_dai_get_widget(dai, substream->stream);
 	swidget = w->dobj.private;
+	pipe_widget = swidget->pipe_widget;
+	pipeline = pipe_widget->private;
```

Comment:
I guess no need to do a safety check that `pipeline->is_backend == true`, right?

Reply:
not needed because we can only ever have a DAI widget set as the playback/capture widget for the BE DAIs